ministryofjustice/laa-claim-non-standard-magistrate-fee-backend
# Crm457 151 add laa reference

## Description of change

- Add laa_reference field to claims
- Add generate_laa_reference method to claims (private) to generate a reference with format LAA-[6 alphanumeric characters]
- Make laa_reference generate when a new claim is created
- Add LAA Reference as field on Claims table and Claim task list
- Order records on claim page by updated_at field

## Link to relevant ticket
[Add functionality to generate and store LAA Reference](https://dsdmoj.atlassian.net/browse/CRM457-151)

## Notes for reviewer

## Screenshots of changes (if applicable)

### Before changes:
<img width="922" alt="image" src="https://github.com/ministryofjustice/laa-claim-non-standard-magistrate-fee-backend/assets/26544538/8798bbf8-34d4-4b8e-9ac6-32b594fd904d">
<img width="922" alt="image" src="https://github.com/ministryofjustice/laa-claim-non-standard-magistrate-fee-backend/assets/26544538/4ced9731-0056-4538-a74a-cbff242bcb51">


### After changes:
<img width="766" alt="image" src="https://github.com/ministryofjustice/laa-claim-non-standard-magistrate-fee-backend/assets/26544538/10753d08-6b8f-4ce4-8f3b-df88999381ef">
<img width="685" alt="image" src="https://github.com/ministryofjustice/laa-claim-non-standard-magistrate-fee-backend/assets/26544538/f596ee0c-542e-4d07-bfd7-a3195417e145">


## How to manually test the feature

Claim reference generation
- Go to claims page
- Click Make a New Claim
- Submit claim type
- Go back to claims page (should see claim there with an LAA reference generated)

Check claims page and claim task list to see laa reference
